### PR TITLE
Argo diff

### PR DESCRIPTION
**Pull request by sachinchauhan23**

### CODE REVIEW

File: targets.go [hunk 1]
```diff
@@ -72,6 +72,10 @@ func ArgoCDDiff() error {
 	if err != nil {
 		return err
 	}
+	err = validateKyvernoPolicies(apps)
```
Comment (AtzeDeVries) [inline targets.go:75]: can we not wrap this in a separate target, like `KubeConform()` (line 34)

Comment (sachinchauhan23) [reply targets.go]: wrapping this under kubeconform cause validation errors as the parameter passed is not defined there.

**Comment by AtzeDeVries** [comment]
I think you should approach this differently.  I think you should mimic the behaviour of for example the `KubeScore()` function. Calls a function which renders templates from information fetched from ArgoCD, then passes the rendered templates on to `kubeScore(paths)` to do the actual work. 

You are now inputting argocdapps into the `validateKyvernoPolicies`. For kyverno to run, we  only require the rendered templates from the app, which is already handeled by the `renderTemplates()` fuction.

I think it is best to create a validateKyvernoPolicies function that just expects a list of rendered templates (paths). 

This will simplify testing a lot and also improve reusability.  

Currently the test fails because it is unable to download the repository where the policyfiles are stored. I would omit this for now and store the policy files in this repo (and also in the repo where the engineering image is build). Create an issue here that adds the task to fix auto downloading the policies.

File: targets_test.go [hunk 2]
```diff
@@ -36,3 +36,21 @@ func TestOKKubeConform(t *testing.T) {
...
 }
+
+// Test for manifest files expected to fail Kyverno policy validation
+// func TestFailedValidateKyverno(t *testing.T) {
+// 	path := "tests/templates/validate-fail/"
+// 	err := validateKyvernoPolicies(path)
+// 	if err == nil {
+// 		t.Fatalf("Expected validation to fail for manifest %s, but it passed", path)
+// 	}
+// }
```
Comment (AtzeDeVries) [inline targets_test.go:47]: why is this commented out?

Comment (sachinchauhan23) [reply targets_test.go]: not sure how this crept in there. i was testing with these changes. uncommented now.

**Review by AtzeDeVries — APPROVED** [review]

**Pull request merged**